backend/routes/boards.js

```diff
@@ -1007,4 +1011,47 @@ router.delete('/:id(\\d+)/remove-members', authJWT, async (req, res) => {
       .json({ board_id: null, members_removed: null, error_message: err });
   }
 });
+
+router.delete('/:id(\\d+)/removeRepository', authJWT, async (req, res) => {
+  const { id } = req.params;
+  const { id: userId } = req.user;
+  const { id: repoId } = req.body;
+
+  if (id === undefined) {
+    return res
+      .status(400)
+      .json({ id: null, title: null, error_message: boardErrorMessages.MISSING_ID });
+  }
+
+  try {
+    await getPermissions(userId, id);
+  } catch (err) {
+    debug(err);
+    return res.status(403).json({ id: null, title: null, error_message: err });
+  }
+
+  let board = null;
```

Comment:
no need to check if board exists. if board does not exists, this results in an error in `getPermssions`